### PR TITLE
CATROID-581 Refactor handling of screen sizes in StageActivity

### PR DESCRIPTION
**Pull request by moberer**
https://jira.catrob.at/browse/CATROID-581

In the StageActivity there exist a few methods which are responsible for resizing the project's resolution in order the fit the mobile device's screen.
This code was extracted into a new class (-> Resolution) and tests were created. Same or similar calculations in other places throughout the code base were refactored to use this new class in order to consolidate the responsibility for resizing in one single place.

*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [X] Include the name of the Jira ticket in the PR’s title
- [X] Include a summary of the changes plus the relevant context
- [X] Choose the proper base branch (*develop*)
- [X] Confirm that the changes follow the project’s coding guidelines
- [X] Verify that the changes generate no compiler or linter warnings
- [X] Perform a self-review of the changes
- [X] Verify to commit no other files than the intentionally changed ones
- [X] Include reasonable and readable tests verifying the added or changed behavior
- [X] Confirm that new and existing unit tests pass locally
- [X] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [X] Stick to the project’s gitflow workflow
- [X] Verify that your changes do not have any conflicts with the base branch
- [X] After the PR, verify that all CI checks have passed
- [X] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by moberer** [comment]
State:
Already implemented centralized class for doing the calculations;
Already used in some places;

TODO: Find other places where the centralized functionality should be used.

**Review by KarlDerMaisbauer — COMMENTED** [review]
Please remove TODO in StageListener.java

**Review by KarlDerMaisbauer — APPROVED** [review]

File: catroid/src/main/java/org/catrobat/catroid/camera/PreviewView.kt [hunk 1]
```diff
@@ -41,31 +40,13 @@ class PreviewView(context: Context) : FrameLayout(context) {
...
         )
     }
-
-    private fun scaleView(imageWidth: Int, imageHeight: Int) {
-        val imageAspectRatio = imageHeight.toFloat() / imageWidth
-        val screenAspectRatio = this.width.toFloat() / this.height
-
-        if (screenAspectRatio < 1) { // portrait mode
-            val scalingFactor = imageAspectRatio / screenAspectRatio
-            if (Build.VERSION.SDK_INT == Build.VERSION_CODES.M) {
```
Comment (becothas) [inline catroid/src/main/java/org/catrobat/catroid/camera/PreviewView.kt:59]: @mseiser do you know if the special behavior for SDK 23(Android M) is needed?

Comment (mseiser) [reply catroid/src/main/java/org/catrobat/catroid/camera/PreviewView.kt:59]: Our MinSDK is still SDK 23, please have a look at the AndroidManifest.xml, regarding SDK requirements

File: catroid/src/main/java/org/catrobat/catroid/utils/Resolution.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+private const val EQUALITY_EPSILON = 0.0000001
+class Resolution @JvmOverloads constructor(
+    val width: Int,
+    val height: Int,
+    val offsetX: Int = 0,
+    val offsetY: Int = 0
+) {
+    val aspectRatio: Float = width.toFloat() / height.toFloat()
+    val isInPortraitMode: Boolean = width < height
+    val isInLandscapeMode: Boolean = !isInPortraitMode
```
[the remark this answers is not in the document]
Comment (moberer) [reply catroid/src/main/java/org/catrobat/catroid/utils/Resolution.kt]: Done;

File: catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java [hunk 3]
```diff
@@ -150,10 +149,7 @@ public class StageListener implements ApplicationListener {
 
 	private StageDialog stageDialog;
 
-	int maxViewPortX = 0;
-	int maxViewPortY = 0;
-	int maxViewPortHeight = 0;
-	int maxViewPortWidth = 0;
+	Resolution maxViewPort = null;
```
[the remark this answers is not in the document]
Comment (moberer) [reply catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java]: Implemented the requested change;

**Review by becothas — APPROVED** [review]
Looks good.

**Comment by Shubhgajj2004** [comment]
Hey, in the file `Resolution.kt`, within the function `testThatAspectRatioRemainsUnchangedAfterResize()`, parentheses are remaining in 'aspectRatio', hence throwing an error. Just refactor '`aspectionRatio`' to '`aspectRatio()`'.

**Pull request merged**